script/main.js

```diff
@@ -8,52 +8,53 @@ const form = document.getElementById('form');
 const inputTitle = document.getElementById('title');
 const inputAuthor = document.getElementById('author');
 
-const store = localStorage.getItem('store') ? JSON.parse(localStorage.getItem('store')) : [];
-
-function removeBook(book) {
-  const bookItems = Array.from(document.getElementsByClassName('book'));
-  Object.keys(bookItems).forEach((key) => {
-    if (book === key) {
-      bookItems[key].remove();
-      store.splice(key, 1);
-      localStorage.setItem('store', JSON.stringify(store));
-    }
-  });
+class Book {
+  constructor() {
+    this.store = localStorage.getItem('store') ? JSON.parse(localStorage.getItem('store')) : [];
+  }
+
+  addBook() {
+    this.store.push({
+      title: inputTitle.value,
+      author: inputAuthor.value,
+    });
+    localStorage.setItem('store', JSON.stringify(this.store));
+    inputTitle.value = '';
+    inputAuthor.value = '';
+  }
+
+  removeBook(book) {
+    const title = book.querySelector('#book-title').innerText;
+    book.remove();
+    this.store = this.store.filter((bookItem) => bookItem.title !== title);
+    localStorage.setItem('store', JSON.stringify(this.store));
+  }
```

Comment:
- [ ] Unfortunately, the refactor introduced a bug to this function, if I have two or more books with the same title, try to remove one and refresh the page or add another book, all the books with that title will be removed, instead of just the targeted one, kindly fix that.

Reply:
Reconstruct script: add bookId.